### PR TITLE
Remove gathering requirements from user environment

### PR DESCRIPTION
**Pull request by jpurusho65**
## Describe your changes and why you are making these changes
Remove gathering module prerequisites from environment using pip freeze command
This is error prone, and different user environment may have different modules, and leads to inconsistent runtime behavior.
This change raises a RequirementsFileMissingError exception if either requirement is not provided as a list parameter or requirement.txt file is missing in the current working directory of the imported python function.

## Related issue number (if any)
ENG 2551

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/utils/function_packaging.py
Comment (kenxu95) [inline sdk/aqueduct/utils/function_packaging.py:81]: What do you think about using a custom RequirementsNotFoundError class instead? I think it may make more sense since requirements can be defined in code (@op) or in a file (requirements.txt)

File: sdk/aqueduct/tests/decorator_test.py
Comment (kenxu95) [inline sdk/aqueduct/tests/decorator_test.py:59]: Can you add the "pytest.raises(..., match=...)" argument so we also confirm the error message here?

File: sdk/aqueduct/tests/decorator_test.py
Comment (kenxu95) [inline sdk/aqueduct/tests/decorator_test.py:63]: Why does this fail? Isn't there a requirements.txt file in `test_files/python_function`? I guess I'm confused as to why there's a requirements.txt file in test_files/python_function at all.

File: sdk/aqueduct/utils/function_packaging.py
Comment (kenxu95) [inline sdk/aqueduct/utils/function_packaging.py:223]: Let's give the user a little bit more guidance here. Can you add that the requirements.txt file should be in the same directory as the function definition? And that you can alternatively add the requirements directly in the @op/@metric/@check decorator.

**Review by kenxu95 — CHANGES_REQUESTED** [review]
Changes look mostly good! Just a couple small requests/questions

Can we also add one integration test that tries to run an @op that doesn't have requirements? Very similar to the unit test but at a higher level to capture user behavior.

File: sdk/aqueduct/tests/decorator_test.py
Comment (jpurusho65) [reply sdk/aqueduct/tests/decorator_test.py]: done

File: integration_tests/backend/setup/flow_with_failure.py
Comment (kenxu95) [inline integration_tests/backend/setup/flow_with_failure.py:19]: Any reason we can't just add `requirements.txt` to `integration_tests/backend/`? This seems like a hassle.

File: integration_tests/no_concurrency/sdk/requirements_test.py
Comment (kenxu95) [inline integration_tests/no_concurrency/sdk/requirements_test.py:49]: Same comment about adding `requirements.txt` to `integration_tests/no_concurrency/sdk`.

File: sdk/aqueduct/decorator.py
Comment (kenxu95) [inline sdk/aqueduct/decorator.py:828]: We raise a `RequirementsFileMissingError` right?

File: sdk/aqueduct/error.py
Comment (kenxu95) [inline sdk/aqueduct/error.py:136]: Sorry it's a super small language nit but I wanted this to just be `RequirementsMissingError` without the "file" part, since requirements can be specified in multiple ways.

File: sdk/aqueduct/utils/function_packaging.py
Comment (kenxu95) [inline sdk/aqueduct/utils/function_packaging.py:221]: nit: "definition. Alternatively, add ..."

**Review by kenxu95 — APPROVED** [review]
Happy to unblock but please address the comments before merging :) It would be best if we could add `requirements.txt` to our test folders, instead of forcing every @op defined in them to have `requirements=[]`.

**Pull request merged**
